can/interfaces/vector/canlib.py

```diff
@@ -217,20 +224,36 @@ def __init__(
         if permission_mask.value == self.mask:
             if fd:
                 self.canFdConf = xlclass.XLcanFdConf()
-                if bitrate:
-                    self.canFdConf.arbitrationBitRate = ctypes.c_uint(bitrate)
+                if timing:
+                    self.canFdConf.arbitrationBitRate = ctypes.c_uint(timing.bitrate)
+                    self.canFdConf.sjwAbr = ctypes.c_uint(timing.sjw)
+                    self.canFdConf.tseg1Abr = ctypes.c_uint(timing.tseg1)
+                    self.canFdConf.tseg2Abr = ctypes.c_uint(timing.tseg2)
+                    if data_timing:
+                        self.canFdConf.dataBitRate = ctypes.c_uint(data_timing.bitrate)
+                        self.canFdConf.sjwDbr = ctypes.c_uint(data_timing.sjw)
+                        self.canFdConf.tseg1Dbr = ctypes.c_uint(data_timing.tseg1)
+                        self.canFdConf.tseg2Dbr = ctypes.c_uint(data_timing.tseg2)
+                    else:
```

Comment:
This `else` clause seems unnecessary - instead why don't we make sure  `data_timing` falls back to the `timing` object